### PR TITLE
mpir: Fix fPIC

### LINKED ISSUE #1764: [package] mpir/3.0.0: fPIC not working
### Package and Environment Details
  * Package Name/Version: **mpir/3.0.0**
  * Operating System+version: **Manjaro 20.0.1**
  * Compiler+version: **Clang 10**
  * Conan version: **conan 1.25.2**
  * Python version: **Python 3.8.2**

### Conan profile
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=10
compiler.libcxx=libstdc++11
compiler.cppstd=17
build_type=Release
python_version=3.7
[options]
[build_requires]
[env]
CC=/usr/bin/clang
CXX=/usr/bin/clang++
CXX_CLANG_TIDY=/usr/bin/clang-tidy
CXX_INCLUDE_WHAT_YOU_USE=/usr/bin/include-what-you-use
Python27_EXECUTABLE=/usr/bin/python2.7
Python37_EXECUTABLE=/usr/bin/python3.7
```

### Steps to reproduce (Include if Applicable)
Build shared library/python module that uses `mpir` as a static library`

### Logs (Include/Attach if Applicable)
<details><summary>Recompile with -fPIC ... Click to expand log</summary>

```
/usr/bin/ld: /home/birger/.conan/data/mpir/3.0.0/_/_/package/7663ba8f8fea4c384def53826e87140212375ef9/lib/libmpir.a(modexact_1c_odd.o): relocation R_X86_64_PC32 against symbol `__gmp_modlimb_invert_table' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
```

</details>

### PR DESCRIPTION
**Pull request by birgerbr**
Add `--with-pic` argument to configure when fPIC option is true.

closes #1764 

Specify library name and version:  **mpir/3.0.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mpir/all/conanfile.py [hunk 1]
```diff
@@ -62,6 +62,8 @@ def _configure_autotools(self):
                 args.extend(['--disable-static', '--enable-shared'])
             else:
                 args.extend(['--disable-shared', '--enable-static'])
+            if self.options.fPIC:
+                args.extend(['--with-pic'])
```
Comment (madebr) [inline recipes/mpir/all/conanfile.py:66]: Doesn't this suffice? https://github.com/conan-io/conan/blob/ed6b33681d1cf25e486518400d0bd845bf5c9aa1/conans/client/build/autotools_environment.py#L326-L327

Comment (madebr) [reply recipes/mpir/all/conanfile.py:66]:
```suggestion
            args.append(["--with-pic" if self.options.get_safe("fPIC") else "--without-pic"])
```
This should fix a potential problem for mingw.

Comment (birgerbr) [reply recipes/mpir/all/conanfile.py:66]: Using the code above (without the brackets) works.
`args.append("--with-pic" if self.options.get_safe("fPIC") else "--without-pic")`

https://github.com/conan-io/conan/blob/ed6b33681d1cf25e486518400d0bd845bf5c9aa1/conans/client/build/autotools_environment.py#L326-L327
does not seem to be sufficient in my case.

**Comment by birgerbr** [comment]
Nice!
Will the CI be triggered to run when I'm accepted to the Early Access Program?

**Comment by uilianries** [comment]
@birgerbr you are in. Because the CI bot checked your PR. Please, add those end empty lines to fix this package and we are good. This error is not part of your PR, but because we added a hook which checks it.

**Pull request merged**